### PR TITLE
Fixes #3 - DynamicListRoute has been deprecated

### LINKED ISSUE #3: DynamicListRoute is deprecated
List route and detail route has been merged into a single action as of [DRF v3.8.0](https://www.django-rest-framework.org/topics/release-notes/#380) and hence the `SingletonRouter` in `base/api/router.py` needs refactoring.

### PR DESCRIPTION
**Pull request by realslimshanky**
DynamicListRoute has been replaced with DynamicRoute
sample_method added to users/api

### CODE REVIEW

File: nexus/base/api/routers.py [hunk 1]
```diff
@@ -29,9 +29,10 @@ class SingletonRouter(routers.SimpleRouter):
         # Dynamically generated list routes.
         # Generated using @list_route decorator
         # on methods of the viewset.
-        routers.DynamicListRoute(
-            url=r'^{prefix}/{methodname}{trailing_slash}$',
-            name='{basename}-{methodnamehyphen}',
+        routers.DynamicRoute(
+            url=r'^{prefix}/{url_path}$',
```
Comment (CuriousLearner) [inline nexus/base/api/routers.py]: In this case,  how do we accomdate `methodname`. Can you pass on the links to docs?

Comment (realslimshanky) [reply nexus/base/api/routers.py]: We don't need to use `methodname` explicitly, `url_path` would suffice. [source](https://www.django-rest-framework.org/api-guide/routers/#customizing-dynamic-routes)

**Comment by CuriousLearner** [comment]
@realslimshanky Seems like you deleted your repository here. Possible to fetch this branch and update this so that I can test? :)

**Comment by CuriousLearner** [comment]
@realslimshanky Any updates on this one?

**Comment by realslimshanky** [comment]
@CuriousLearner I cannot seem to push to this PR. So in order to move forward use this patch [fixing-dynamic-route.txt](https://github.com/CuriousLearner/nexus/files/2416767/fixing-dynamic-route.txt) for testing

**Pull request closed without merging**